### PR TITLE
[8.x] Adds Str helper for encoding to URL Query

### PR DESCRIPTION
**Pull request by DarkGhostHunter**
## What?

This allows to encode strings to URL query strings. 

```php
use Illuminate\Support\Str;

$query = Str::urlQuery('This sign "=" is / cool!') // This+sign+%22%3D%22+is+%2F+cool%21

return redirect()->to('https://myapp.com/?check=' . $query);
```

## Why?

Because sometimes you have to deal with query strings from users in a simple way.

## BC?

None, it's additive.

## Notes?

Well, you may have a point using `urlencode(Str::lower($string))` but I think `urlQuery()` is more descriptive.

### CODE REVIEW

File: src/Illuminate/Support/Str.php [hunk 1]
```diff
@@ -818,6 +818,17 @@ public static function ucfirst($string)
...
 
+    /**
+     * Encodes the string to a URL-query-safe string.
+     *
+     * @param  $string
+     * @return string
+     */
+    public static function urlQuery($string)
+    {
+        return urlencode(static::lower($string));
```
Comment (garygreen) [inline src/Illuminate/Support/Str.php:829]: Why lowercase? Seems opinionated

Comment (DarkGhostHunter) [reply src/Illuminate/Support/Str.php:829]: You're totally right, shouldn't lower case, but may be I should add the option to normalize the string to lowercase.

File: src/Illuminate/Support/Str.php [hunk 2]
```diff
@@ -818,6 +818,17 @@ public static function ucfirst($string)
...
 
+    /**
+     * Encodes the string to a URL-query-safe string.
+     *
+     * @param  $string
+     * @return string
+     */
+    public static function urlQuery($string)
+    {
+        return urlencode(trim($string));
```
Comment (GrahamCampbell) [inline src/Illuminate/Support/Str.php:829]: I don't think the trimming makes sense here either. What's the reason to not allow whitespace around the string. I think this method in the core is probably unnecessary. Just define a macro in your app that does what you need, or just call urlencode.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Pull request closed without merging**